src/StatusQ/Controls/StatusBaseInput.qml

```diff
@@ -0,0 +1,113 @@
+import QtQuick 2.14
+
+import StatusQ.Controls 0.1
+import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
+
+
+Item {
+    id: statusBaseInput
+
+    property alias inputMethodHints: edit.inputMethodHints
+
+    property alias selectedText: edit.selectedText
+    property alias selectedTextColor: edit.selectedTextColor
+    property alias selectionStart: edit.selectionStart
+    property alias selectionEnd: edit.selectionEnd
+
+    property alias color: edit.color
+    property alias font: edit.font
+    property alias verticalAlignmet: edit.verticalAlignment
+    property alias horizontalAlignment: edit.horizontalAlignment
+
+    property alias placeholderText: placeholder.text
+    property alias placeholderTextColor: placeholder.color
+    property alias placeholderFont: placeholder.font
+
+    property real leftPadding: 16
+    property real rightPadding: 16
+    property real topPadding: 11
+    property real bottomPadding: 11
+
+    implicitWidth: 200
+    height: 40
+
+    clip: true
+
+    Rectangle {
+        anchors.fill: parent
+        color: Theme.palette.baseColor2
+        radius: 8
+    }
+
+    Flickable {
+        id: flick
+
+        anchors.fill: parent
+        anchors.leftMargin: statusBaseInput.leftPadding
+        anchors.rightMargin: statusBaseInput.rightPadding + clearButtton.width
+        anchors.topMargin: statusBaseInput.topPadding
+        anchors.bottomMargin: statusBaseInput.bottomPadding
+        contentWidth: edit.paintedWidth
+        contentHeight: edit.paintedHeight
+        clip: true
```

Comment:
Couple of things here:

1. Is it possible to turn this off from the outside? Many of the inputs are just single line inputs (search input, password input, etc). Having the base input being scrollable/flickable by default seems off. The default should probably be that it's just a single line input. We can even split this up into separate components if needed (e.g. StatusBaseInput, StatusBaseTextArea).

2. The whole flickable thing doesn't seem to work well. When entering information and hitting SHIFT+Enter and then continuing typing, this is what it looks like:

<img width="200" alt="Screenshot 2021-06-29 at 09 59 29" src="https://user-images.githubusercontent.com/445106/123760092-d0015200-d8c0-11eb-8955-18a07f3cb663.png">

Notice how the entered text isn't fully visible, users can't see the cursor anymore. either if they just add a few more lines.

Reply:
This seems to behave much better now.
One thing however, is still unexpected:

If it's `multiline: true` and I enter some value it looks like this:

<img width="223" alt="Screenshot 2021-07-05 at 17 41 10" src="https://user-images.githubusercontent.com/445106/124495260-305c2c00-ddb8-11eb-9fb6-c9fcc8d741e6.png">

When I then hit Shift+ Enter (or just Enter) to get to the next line, it looks like this:

<img width="218" alt="Screenshot 2021-07-05 at 17 42 01" src="https://user-images.githubusercontent.com/445106/124495361-4f5abe00-ddb8-11eb-840c-a5a95cc7ee67.png">

Basically the first line I've entered disappears because the input value only renders the last line (**this might be expected**).

However, if I do the same thing again, all of a sudden the input starts to grow:
<img width="220" alt="Screenshot 2021-07-05 at 17 43 02" src="https://user-images.githubusercontent.com/445106/124495482-744f3100-ddb8-11eb-8a27-75a7c3ca5b7a.png">

This seems less unexpected to me, but I believe we should be doing either one or the other behaviour (either the height is fixed and text value becomes scrollable, or it always grows [unless a fixed height was supplied from the outside]).

What do you think?

Reply:
sorry that is size bug.

Reply:
I support this:
> We can even split this up into separate components if needed (e.g. StatusBaseInput, StatusBaseTextArea).
Based on my prior experience it is the best approach.

Reply:
Happy to revisit this. For now though, I'd like us to land **some** input component that really just renders an input because so many other components depend on this.